### ISSUE TITLE
SecretPad组件库中是否有神经网络类模型训练和模型预测组件

### ISSUE THREAD

**Issue by zhangyufenggit** [secretflow/secretpad#269]
SecretPad组件库中是否有神经网络类模型训练和模型预测组件

**Comment by zhangyufenggit** [comment]
> 没有，secretpad所有组件可以参考[官网组件说明](https://www.secretflow.org.cn/zh-CN/docs/secretpad-all-in-one/v1.11.0b1/components_instruction)

我看这个文档中咱们SPU中是有神经网络的，理论上能否按照组件扩展标准封装成一个组件呢？
https://www.secretflow.org.cn/zh-CN/docs/secretflow/v1.12.0b0/tutorial/nn_with_spu

**Comment by zhangyufenggit** [comment]
> 是的，secretpad定义的有nn的组件，但是需要使用[secretflow的FULL版本镜像](https://www.secretflow.org.cn/zh-CN/docs/secretflow/v1.12.0b0/getting_started/installation#option-2-from-docker)，而非lite镜像，同时修改下kuscia的注册的镜像，然后将[application.yaml隐藏的组件](https://github.com/secretflow/secretpad/blob/cc857f70eefd2f9f301246c8cd50f65b4a2442e9/config/application.yaml#L198)打开，看下该组件是否满足 如果不满足，可以二开SecretFlow来自己定义新的组件

老师，在all-in-one部署方式下，【使用secretflow的FULL版本镜像】、【修改kuscia的注册的镜像】具体操作步骤是什么呢？谢谢

**Comment by zimu-yuxi** [comment]
1.可以看下现在secretflow镜像名中有-lite，去掉就可以，可以docker pull来拉
2.镜像注册参考[这里](https://www.secretflow.org.cn/zh-CN/docs/kuscia/v0.15.0b1/development/register_custom_image)

**Comment by zhangyufenggit** [comment]
> 1.可以看下现在secretflow镜像名中有-lite，去掉就可以，可以docker pull来拉 2.镜像注册参考[这里](https://www.secretflow.org.cn/zh-CN/docs/kuscia/v0.15.0b1/development/register_custom_image)

老师，目前我已经注册好secretflow的full版本镜像，组件库也已经可以看到该组件，但实际运行的时候报错，麻烦帮忙看下这个环境变量在哪里配置呢，具体调整哪个yaml文件呢？

![Image](https://github.com/user-attachments/assets/466ce9cd-9571-4c56-86d9-0743670a8726)

**Comment by zimu-yuxi** [comment]
参考[此处](https://www.secretflow.org.cn/zh-CN/docs/kuscia/v0.15.0b1/reference/concepts/appimage_cn#appimage-ref)，尝试在appimage的env中增加

**Comment by zhangyufenggit** [comment]
前置工作都已经完成了，目前可以看到神经网络组件了，但运行时报错，附件为all-in-one部署的两个节点相关日志，node148为发起方，也是Y方，麻烦帮忙看下是什么问题，谢谢！

[node148.tar.gz](https://github.com/user-attachments/files/21075516/node148.tar.gz)

[node149.tar.gz](https://github.com/user-attachments/files/21075517/node149.tar.gz)

**Comment by zimu-yuxi** [comment]
确认下，secretpad和使用的Secretflow版本分别是多少

**Comment by zimu-yuxi** [comment]
方便提供下组件配置详细的截图吗？
看日志报错很多，有输入的特征数和模型期望的特征数不一致

**Comment by zhangyufenggit** [comment]
1、特征列：

<img width="811" height="590" alt="Image" src="https://github.com/user-attachments/assets/e9165aca-a2b4-4552-b4ad-58d8970cc32e" />
2、标签列：label
3、模型：（组件默认，未修改）
# pre imported:# import tensorflow as tf# from tensorflow import Module, keras# from tensorflow.keras import Model, layers# from tensorflow.keras.layers import Layer# from secretflow.ml.nn import applications as appsdef create_base_model(input_dim, output_dim):    model = keras.Sequential(        [            keras.Input(shape=input_dim),            layers.Dense(100, activation="relu"),            layers.Dense(output_dim, activation="relu"),        ]    )    return modeldef create_fuse_model(input_dim):    input_layers = [keras.Input(input_dim), keras.Input(input_dim)]    merged_layer = layers.concatenate(input_layers)    fuse_layer = layers.Dense(64, activation='relu')(merged_layer)    output = layers.Dense(1, activation='sigmoid')(fuse_layer)    return keras.Model(inputs=input_layers, outputs=output)hidden_size = 64fit(    client_base=create_base_model(12, hidden_size),    server_base=create_base_model(4, hidden_size),    server_fuse=create_fuse_model(hidden_size),)
4、其它也都是默认参数，未修改：

<img width="1082" height="817" alt="Image" src="https://github.com/user-attachments/assets/13923dce-016d-4a19-bc03-b75ca7ec47db" />

**Comment by zhangyufenggit** [comment]
是不是和使用的数据有关呢？咱们有验证这个组件的测试数据么？是否可以发给我，我验证一下，谢谢。

**Comment by zhangyufenggit** [comment]
麻烦问下这个问题有进展么，谢谢

**Comment by wangzul** [comment]
尝试减少一些特征数量尝试下 4/12 。

**Comment by zhangyufenggit** [comment]
> 尝试减少一些特征数量尝试下 4/12 。

已经尝试减少特征，依然有问题。是否可以通过升级SecretFlow版本解决这个问题？
